### PR TITLE
Feature - User Login

### LINKED ISSUE #2: Basic User Login

### PR DESCRIPTION
**Pull request by vdevic01**
Closes #2

### CODE REVIEW

File: DockerHubBackend/DockerHubBackend/Data/DatabaseContextSeed.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿using DockerHubBackend.Models;
+using Microsoft.AspNetCore.Identity;
+
+namespace DockerHubBackend.Data
+{
+    public static class DatabaseContextSeed
```
Comment (SelenaMilutin) [inline DockerHubBackend/DockerHubBackend/Data/DatabaseContextSeed.cs:6]: What is this Class used for? Is this for the superuser? Why are two users saved?

File: DockerHubBackend/DockerHubBackend/Dto/Response/LoginResponseWithJwt.cs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+﻿namespace DockerHubBackend.Dto.Response
+{
+    public class LoginResponseWithJwt
+    {
+        public LoginResponse Response { get; set; }
+        public string Jwt { get; set; }
```
Comment (SelenaMilutin) [inline DockerHubBackend/DockerHubBackend/Dto/Response/LoginResponseWithJwt.cs:6]: I think it's nice you separated the LoginResponse from jwt

File: DockerHubBackend/DockerHubBackend/Repository/Utils/CrudRepository.cs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            {
+                _entities.Update(entity);
+                await _context.SaveChangesAsync();
+                return entity;
+            }            
+            return null;
+        }
+
+    }
+}
```
Comment (SelenaMilutin) [inline DockerHubBackend/DockerHubBackend/Repository/Utils/CrudRepository.cs:60]: Very nice, very nice

File: DockerHubBackend/DockerHubBackend/appsettings.json [hunk 4]
```diff
@@ -8,5 +8,10 @@
   "AllowedHosts": "*",
   "ConnectionStrings": {
     "DefaultConnection": "Host=localhost;Port=5432;Database=uks-database;Username=admin;Password=admin"
+  },
+  "Jwt": {
+    "Key": "D47C7FD6E2D7DVM20VSX1YVP6289QBWN62MX33FIUH",
```
Comment (SelenaMilutin) [inline DockerHubBackend/DockerHubBackend/appsettings.json:13]: Should we add this values to an env file?

File: DockerHubBackend/DockerHubBackend/Security/JwtHelper.cs [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+        public JwtHelper(IConfiguration configuration)
+        {
+            _key = configuration["Jwt:Key"];
+            _issuer = configuration["Jwt:Issuer"];
+            _expiration = Convert.ToInt32(configuration["Jwt:Expiration"]);
+        }
+        public string GenerateToken(string role, string userId, DateTime? lastPasswordChange)
+        {
+            string lastPasswordChangeStr = lastPasswordChange != null ? lastPasswordChange.ToString() : DateTime.MinValue.ToString("o");
```
Comment (SelenaMilutin) [inline DockerHubBackend/DockerHubBackend/Security/JwtHelper.cs:22]: @miloss01 asks what is this field for

**Review by SelenaMilutin — APPROVED** [review]

File: DockerHubBackend/DockerHubBackend/Security/JwtHelper.cs (hunk 5 above)
Comment (miloss01) [reply DockerHubBackend/DockerHubBackend/Security/JwtHelper.cs:22]: namesta me da ispadnem glup

**Comment by miloss01** [comment]
Should we merge PR without frontend side?

File: DockerHubBackend/DockerHubBackend/Controllers/AuthController.cs [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+            _authService = authService;
+            _configuration = configuration;
+        }
+
+        [HttpPost("login")]
+        public async Task<IActionResult> Login([FromBody] LoginCredentialsDto credentials)
+        {
+            var response = await _authService.Login(credentials);
+            var tokenExpiration = Convert.ToInt32(_configuration["JWT:Expiration"]);
+            Response.Cookies.Append("AuthToken", response.Jwt, new CookieOptions
```
Comment (miloss01) [inline DockerHubBackend/DockerHubBackend/Controllers/AuthController.cs:29]: Is JWT passed using cookies? Did you consider passing it as a Authorization header?

File: DockerHubBackend/DockerHubBackend/Data/DatabaseContextSeed.cs (hunk 1 above)
Comment (vdevic01) [reply DockerHubBackend/DockerHubBackend/Data/DatabaseContextSeed.cs:6]: It's just a script to prepopulate database for development purposes.

File: DockerHubBackend/DockerHubBackend/appsettings.json (hunk 4 above)
Comment (vdevic01) [reply DockerHubBackend/DockerHubBackend/appsettings.json:13]: Yes, both jwt key and database connection string should be environment variables. I had a trouble making it work so I just left it like this for now.

File: DockerHubBackend/DockerHubBackend/Security/JwtHelper.cs (hunk 5 above)
Comment (vdevic01) [reply DockerHubBackend/DockerHubBackend/Security/JwtHelper.cs:22]: Let's say access token lasts for 1 day. In mean time a user changes the password. "LastPasswordChange" makes sure that token is not valid anymore.

Comment (miloss01) [reply DockerHubBackend/DockerHubBackend/Security/JwtHelper.cs:22]: Should we implement password change functionality (I don't see it in specs)?

File: DockerHubBackend/DockerHubBackend/Controllers/AuthController.cs (hunk 6 above)
Comment (vdevic01) [reply DockerHubBackend/DockerHubBackend/Controllers/AuthController.cs:29]: I considered that option, but cookie with HttpOnly, Secure and SameSite properties set is much more secure.

**Comment by vdevic01** [comment]
> Should we merge PR without frontend side?

I will make a new issue for frontend part. I want this PR on dev branch ASAP so you can use base classes that are included in it.

**Pull request merged**